src/main/java/buildcraftAdditions/tileEntities/TileLavaCoil.java

```diff
@@ -68,50 +78,30 @@ public void updateEntity() {
 
 	@Override
 	public int fill(ForgeDirection from, FluidStack resource, boolean doFill) {
-		if (resource.getFluid().getID() == FluidRegistry.LAVA.getID()) {
-			int amount = 3000;
-			if (doFill) {
-				if (resource.amount > 3000 - lava)
-					amount = 3000 - lava;
-				else if (resource.amount < 3000 - lava)
-					amount = resource.amount;
-				lava = lava + amount;
-				return amount;
-			} else {
-				if (resource.amount > 3000 - lava)
-					amount = 3000 - lava;
-				else if (resource.amount < 3000 - lava)
-					amount = resource.amount;
-				return amount;
-			}
-		}
-		return 0;
+		return tank.fill(resource, doFill);
 	}
 
 	@Override
 	public FluidStack drain(ForgeDirection from, FluidStack resource, boolean doDrain) {
+		if (resource != null) {
```

Comment:
wouldn't it be easier if you do this in the tank? also you can just return null here as nobody uses this as far as i know